Source/Soot.cpp

```diff
@@ -0,0 +1,179 @@
+
+
+#ifdef PELEC_USE_SOOT
+#include "PeleC.H"
+#include "SootModel.H"
+#include "SootModel_derive.H"
+#include <Transport.H>
+
+void
+PeleC::setSootIndx()
+{
+  SootComps sc;
+  sc.qRhoIndx = QRHO;
+  sc.qTempIndx = QTEMP;
+  sc.qSpecIndx = QFS;
+  sc.qSootIndx = QFSOOT;
+  sc.rhoIndx = URHO;
+  sc.engIndx = UEDEN;
+  sc.specIndx = UFS;
+  sc.sootIndx = UFSOOT;
+  soot_model->setIndices(sc);
+}
+
+void
+PeleC::addSootDerivePlotVars(
+  amrex::DeriveList& derive_lst, const amrex::DescriptorList& desc_lst)
+{
+  // Add in soot variables
+  amrex::Vector<std::string> sootNames = {"rho_soot", "sum_rho_soot"};
+  derive_lst.add(
+    "soot_vars", amrex::IndexType::TheCellType(), sootNames.size(), sootNames,
+    soot_genvars, amrex::DeriveRec::TheSameBox);
+  derive_lst.addComponent("soot_vars", desc_lst, State_Type, URHO, 1);
+  derive_lst.addComponent(
+    "soot_vars", desc_lst, State_Type, UFSOOT, NUM_SOOT_MOMENTS + 1);
+
+  // Variables associated with the second mode (large particles)
+  amrex::Vector<std::string> large_part_names = {"NL", "soot_V_L", "soot_S_L"};
+  derive_lst.add(
+    "soot_large_particles", amrex::IndexType::TheCellType(),
+    large_part_names.size(), large_part_names, soot_largeparticledata,
+    amrex::DeriveRec::TheSameBox);
+  derive_lst.addComponent(
+    "soot_large_particles", desc_lst, State_Type, UFSOOT, NUM_SOOT_MOMENTS + 1);
+}
+
+void
+PeleC::construct_old_soot_source(amrex::Real time, amrex::Real dt)
+{
+  old_sources[soot_src]->setVal(0.0);
+  if (!add_soot_src) {
+    return;
+  }
+  amrex::MultiFab& S_old = get_old_data(State_Type);
+
+  int ng = 0; // None filled
+
+  PeleC::fill_soot_source(time, dt, S_old, *old_sources[soot_src], ng);
+
+  old_sources[soot_src]->FillBoundary(geom.periodicity());
+}
+
+void
+PeleC::construct_new_soot_source(amrex::Real time, amrex::Real dt)
+{
+  new_sources[soot_src]->setVal(0.0);
+  if (!add_soot_src) {
+    return;
+  }
+  amrex::MultiFab& S_new = get_new_data(State_Type);
+
+  int ng = 0;
+
+  PeleC::fill_soot_source(time, dt, S_new, *new_sources[soot_src], ng);
+}
+
+void
+PeleC::fill_soot_source(
+  amrex::Real time,
+  amrex::Real dt,
+  amrex::MultiFab& state,
+  amrex::MultiFab& soot_src,
+  int ng)
+{
+  BL_PROFILE("PeleC::fill_soot_source()");
+
+  auto const& fact =
+    dynamic_cast<amrex::EBFArrayBoxFactory const&>(state.Factory());
+  auto const& flags = fact.getMultiEBCellFlagFab();
+
+#ifdef _OPENMP
```

Comment:
replace with `AMREX_USE_OMP` (this is necessary for Ascent). Even better but a little more work is to use the new ParallelFor way of doing things, see https://github.com/AMReX-Combustion/PeleC/blob/development/Source/External.cpp#L58. THis removes the need for an mfiter loop and a parallelfor loop.